server/server.js

```diff
@@ -1,15 +1,98 @@
 const express = require("express");
 const app = express();
 const port = process.env.PORT || 5000;
+const exampledata = require('../exampleresponse.json')
+const {Pool} = require('pg');
+const cors = require ('cors')
+require('dotenv').config({path:'./.env'})
 
+const db = new Pool({
+  user: process.env.DB_USERNAME,
+  host: process.env.DB_HOST,
+  database: process.env.DB_DATABASE,
+  password: process.env.DB_PASSWORD,
+  port: process.env.DB_PORT,
+});
+
+app.use(express.json()); 
 app.listen(port, () => console.log(`Listening on port ${port}`));
 
 // Store and retrieve your videos from here
 // If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+let videos = exampledata;
 
 // GET "/"
 app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+  const order = req.query.order || "desc";
+
+  // Sort videos based on the 'order' parameter
+  let sortedVideos = videos;
+  if (order === "asc") {
+    sortedVideos = videos.slice().sort((a, b) => a.votes - b.votes);
+  } else if (order === "desc") {
+    sortedVideos = videos.slice().sort((a, b) => b.votes - a.votes);
+  }
+
+  res.json(sortedVideos);
+});
+
+app.post("/", (req, res) => {
+  const { title, url } = req.body;
+
+  // Validate title and URL
+  if (!title || !url) {
+    return res.status(400).json({ result: "failure", message: "Title and URL are required." });
+  }
+
+  // Generate a unique ID for the new video
+  const id = nextVideoId++;
```

Comment:
Where is nextVideoId defined? Can you talk through how this code works / what it does? What's the scope of `nextVideoId`?